src/datepicker/datepicker.ts

```diff
@@ -281,6 +313,7 @@ export class NgbDatepicker implements OnDestroy,
       const oldFocusedDate = this.model ? this.model.focusDate : null;
 
       this.model = model;
+      this.publicState = Object.create({});
```

Comment:
You're still kind of doing `Object.assign()` every time change detection runs with `datepicker.state`, ex in 

```html
 {{ datepicker.state.maxDate }}
```

I still don't understand why you're just not doing this directly: 

```ts
this.publicState = {
  minDate: this.model.minDate,
  // ... etc ...
}
```